dangerfile.js

```diff
@@ -18,9 +18,49 @@ const {danger, warn, fail} = require('danger')
 
 // Danger.js documentation: http://danger.systems/js/
 
+// Check for PRs having a [ ] at the beginning of their title.
+
 if (!danger.github.pr.title.startsWith("[")) {
   warn('This PR title does not include an affected component.'
     + ' For example: "[SomeComponent] Title. If something is affecting multiple components and the'
     + ' change can\'t be broken up into multiple PRs, then the common trait of the change could be'
     + ' used in the PR title. E.g. `[documentation]`.');
 }
+
+// Check for PRs that modify multiple components.
+
+let all_files = danger.git.modified_files
+  .concat(danger.git.deleted_files)
+  .concat(danger.git.created_files);
+
+// Extract any component source modifications.
+let component_files = all_files.filter(function(path) {
+  return path.startsWith('components/') && path.indexOf('/src') >= 0;
+});
+
+// Generate a list of unique components that have been modified.
+let components = Array.from(new Set(component_files.map(function(path) {
+  var path_parts = path.split('/');
+  path_parts.splice(0, 1); // Drop the components/ prefix.
+  
+  var component_path = [];
+  // Convention: all-lower-case names are directories, e.g. `components/private/`.
+  // Capitalized names are components, e.g. `BottomNavigation`.
+  // We want to get the full path up to and including the first component folder.
+  while (path_parts[0][0] == path_parts[0][0].toLowerCase()) {
```

Comment:
If we're checking for "all lowercase", should we lowercase the string and check that instead of only the first character?

Reply:
We only care if the first character is lowercase — the convention is "lower-case first letter is a folder", "upper-case first letter is a component".

Reply:
That's what I thought, but the comment referred to "all-lower-case names".